lnrpc/routerrpc/router_server.go

```diff
@@ -1189,14 +1193,19 @@ func getPair(amtMsat lnwire.MilliSatoshi, amtSat btcutil.Amount,
 	)
 
 	switch {
+	// If a timestamp and amount if provided, return those values.
 	case timeSet && amountSet:
 		return amt, time.Unix(timestamp, 0), nil
 
-	case timeSet && !amountSet:
+	// Return an error if it does have a timestamp without an amount, and
+	// it's not expected to be a failure.
+	case !isFailure && timeSet && !amountSet:
```

Comment:
Same here, should we update the error message to say "... requires non-zero amount for success pairs"?